library/modules/Items.cpp

```diff
@@ -2270,6 +2270,74 @@ bool Items::markForTrade(df::item *item, df::building_tradedepotst *depot) {
     return true;
 }
 
+bool Items::canMelt(df::item *item)
+{
+    CHECK_NULL_POINTER(item);
+
+    switch(item->getType()) {
+        // these are not meltable, even if made from metal
+        case df::item_type::CORPSE:
+        case df::item_type::CORPSEPIECE:
+        case df::item_type::REMAINS:
+        case df::item_type::FISH:
+        case df::item_type::FISH_RAW:
+        case df::item_type::VERMIN:
+        case df::item_type::PET:
+        case df::item_type::FOOD:
+        case df::item_type::EGG:
+        case df::item_type::BAR:
+            return false;
+        default:
+            break;
+    }
```

Comment:
Out of curiosity, where did this section come from? Looks new.

Reply:
This was posted by @ab9rf [on Discord](https://discord.com/channels/793331351645323264/807443277132595211/1173979579409506367). My understanding is that this is the result of reverse engineering. 

That is, except the `BAR` condition, which the game seems to allow but which seems somewhat pointless and was ruled out by the check in `logistics`.

Reply:
suggest replacing the body of the switch with
```
    using namespace df::enums::item_type;
    case CORPSE:
...
```

imo, the scoped `using namespace` statement makes the switch easier to read

iirc, this is from the reverse-engineering of `itemst::meltable()`, which is non-virtual and thus not available for us to use directly

Reply:
Indeed, there is a global `using namespace df::enums;`, and the predominant form is `item_type::CORPSE`, so I would follow that.